File: solutions/java/acronym/1/src/main/java/Acronym.java
import java.util.Arrays;
import java.util.stream.*;

class Acronym {

    private final String acronym;
    Acronym(String phrase) {
        this.acronym = Arrays.asList(phrase.split("[ _-]+")).stream()
            .map(e -> e.toUpperCase().substring(0, 1))
            .collect(Collectors.joining(""));
    }

    String get() {
        return acronym;
    }

}

File: solutions/javascript/acronym/1/acronym.js
export const parse = (phrase) => {
    return phrase
        .split(new RegExp("[ ]|-", "g"))
        .map(e => e.replace(new RegExp("[^a-z]", "gi"), ""))
        .filter(Boolean)
        .map(e => e[0].toUpperCase())
        .join('');
};